File: python/classes/inmostore.client.py
# -*- coding: utf-8 -*-
"""
Este módulo implementa una clase de Python para consumir el API de Portal Services.
Reemplaza las funcionalidades del plugin de Nuxt.js para su uso en entornos de servidor.
"""

import requests

class Website:
    """
    Contiene las rutinas para consumir el API de InmoStore.
    """

    # Definiciones de tipo para la documentación
    # (No son clases, solo referencias para la claridad)
    # ---
    # :type Operacion: dict
    # :param Operacion:
    #     codigo (int): Código de la operación.
    #     newItemId (str): ID del nuevo elemento creado (GUID).
    #     affectedRows (int): Número de filas afectadas por la operación.
    #     mensaje (str): Mensaje de la operación.
    #     detalle (str): Detalle de la operación.
    #
    # :type Elemento: dict
    # :param Elemento:
    #     elementoId (int): ID del elemento.
    #     padre (int): ID del elemento padre.
    #     nombre (str): Nombre del elemento.
    #     tooltiop (str): Tooltip del elemento.
    #     valor (str): Valor del elemento.
    #     defaultValue (str): Valor por defecto del elemento.
    #     status (int): Estado del elemento (0: Inactivo, 1: Activo).
    #
    # :type Escena: dict
    # :param Escena:
    #     escenaId (str): ID de la escena (GUID).
    #     empresaId (str): ID de la empresa a la que pertenece la escena (GUID).
    #     modo (int): Tipo de escena (0: Inmueble, 1: Proyecto).
    #     objecto (str): ID del objeto asociado a la escena (GUID).
    #     titulo (str): Nombre de la escena.
    #     principal (bool): Indica si es la escena principal.
    #     imageUrl (str): URL de la imagen de la escena.
    #     status (int): Estado de la escena (0: Inactiva, 1: Activa).
    #
    # :type EscenaPin: dict
    # :param EscenaPin:
    #     pinId (str): ID del pin (GUID).
    #     empresaId (str): ID de la empresa a la que pertenece el pin (GUID).
    #     escenaId (str): ID de la escena a la que pertenece el pin (GUID).
    #     etiqueta (str): Texto a mostrar del pin.
    #     pitch (str): ...
    #     yaw (str): ...
    #     target (str): ID de la escena a la que apunta el hotspot (GUID).
    #     status (int): Estado del pin (0: Inactivo, 1: Activo).
    #
    # :type Imagen: dict
    # :param Imagen:
    #     imageId (str): ID de la imagen (GUID).
    #     empresaId (str): ID de la empresa a la que pertenece la imagen (GUID).
    #     modo (int): Modo de la imagen (0: Inmueble, 1: Proyecto).
    #     objectId (str): ID del objeto asociado a la imagen (GUID).
    #     titulo (str): Título de la imagen.
    #     imageCaption (str): Descripción de la imagen.
    #     imageUrl (str): URL de la imagen.
    #     status (int): Estado de la imagen (0: Inactiva, 1: Activa).
    #
    # :type Inmueble: dict
    # :param Inmueble:
    #     (contiene muchas propiedades, ver el código JavaScript original)
    #
    # :type ComboItem: dict
    # :param ComboItem:
    #     texto (str): Texto a mostrar del item.
    #     valor (str): Valor del item.
    #
    # :type CaracteristicasInmueble: dict
    # :param CaracteristicasInmueble:
    #     (contiene muchas propiedades, ver el código JavaScript original)
    #
    # :type PropiedadInmueble: dict
    # :param PropiedadInmueble:
    #     (contiene muchas propiedades, ver el código JavaScript original)
    #
    # :type Proyecto: dict
    # :param Proyecto:
    #     (contiene muchas propiedades, ver el código JavaScript original)
    #
    # :type Modelo: dict
    # :param Modelo:
    #     (contiene muchas propiedades, ver el código JavaScript original)
    #
    # :type getElementoResult: dict
    # :param getElementoResult:
    #     Operacion (Operacion): Resultado de la operación.
    #     resultado (Elemento): Detalle del elemento consultado.
    #
    # ... y así sucesivamente para todas las estructuras de datos.

    def __init__(self, api_id: str, api_key: str):
        """
        Inicializa la clase.

        :param api_id: El ID de la API.
        :param api_key: La clave de la API.
        """
        print("InmoStore para Python")

        self.api_id = api_id
        self.api_key = api_key
        self.api_root = "https://inmostore-api.psweb.me/"
        self.api_version = None

    def _fetch(self, url: str):
        """
        Realiza una llamada HTTP asíncrona.

        Este es un método privado que se encarga de la lógica de la petición
        y el manejo básico de errores.

        :param url: La URL de la API a la que se va a llamar.
        :return: El JSON de la respuesta.
        :raises requests.exceptions.RequestException: Si ocurre un error de red o de HTTP.
        """
        try:
            response = requests.get(url, timeout=10)
            response.raise_for_status()  # Lanza una excepción para códigos de estado 4xx/5xx
            return response.json()
        except requests.exceptions.RequestException as e:
            print(f"Error en la llamada a la API: {e}")
            raise

    def init(self):
        """
        Inicia la conexión con el servidor y valida la versión.

        Nota: A diferencia de la versión de Nuxt.js, este código no puede
        redirigir la página ni modificar el HTML. Simplemente imprime un
        mensaje en la consola.
        """
        if not self.api_key:
            print("*** No se ha detectado la configuración del sitio web. ***")
        else:
            try:
                response = self._fetch(f"{self.api_root}sys/version/{self.api_key}")
                self.api_version = response.get("Resultado")
                print(f"Versión {self.api_version}")
            except requests.exceptions.RequestException as e:
                print(f"Error al conectar con el servidor de la API: {e}")

    def get_current_version(self) -> str:
        """
        Obtiene la versión actual del API.

        :returns: La versión del API.
        """
        return self.api_version

    def get_elemento(self, id: int) -> dict:
        """
        Obtiene el detalle de un elemento, consultado por su ID.

        :param id: ID del elemento a consultar.
        :returns: Detalle del elemento consultado.
        """
        return self._fetch(f"{self.api_root}elementos/get/{id}/{self.api_key}")

    def get_empresa(self, id: str) -> dict:
        """
        Obtiene el detalle de la empresa.

        :param id: ID de la empresa a consultar.
        :returns: Detalle de la empresa consultada.
        """
        return self._fetch(f"{self.api_root}empresas/get/{id}/{self.api_key}")

    def list_escenas(self, empresa_id: str, modo: int, object_id: str) -> dict:
        """
        Obtiene la lista de escenas activas de un objeto.

        :param empresa_id: ID de la empresa a consultar.
        :param modo: Modo de la escena (0: Inmueble, 1: Proyecto).
        :param object_id: ID del objeto asociado a la escena.
        :returns: Lista de escenas activas del objeto.
        """
        return self._fetch(f"{self.api_root}escenas/active/{empresa_id}/{modo}/{object_id}/{self.api_key}")

    def list_escena_pins(self, escena_id: str) -> dict:
        """
        Obtiene la lista de pins de la escena actual.

        :param escena_id: ID de la escena a consultar.
        :returns: Lista de pins de la escena consultada.
        """
        return self._fetch(f"{self.api_root}pins/list/{escena_id}/{self.api_key}")

    def get_embeded_image_url(self, image_id: str) -> str:
        """
        Obtiene la URL de una imagen para embeber, basado en su id.

        :param image_id: ID de la imagen a consultar.
        :returns: URL de la imagen a embeber.
        """
        return f"{self.api_root}images/embed/{image_id}/{self.api_key}"

    def list_imagenes(self, modo: int, object_id: str) -> dict:
        """
        Obtiene el listado de imágenes de un objeto.

        :param modo: Modo de la imagen (0: Inmueble, 1: Proyecto).
        :param object_id: ID del objeto asociado a la imagen.
        :returns: Lista de imágenes del objeto consultado.
        """
        return self._fetch(f"{self.api_root}imagenes/list/{modo}/{object_id}/{self.api_key}")

    def get_imagen_portada(self, inmueble_id: str) -> dict:
        """
        Obtiene la imagen de portada de un inmueble.

        :param inmueble_id: ID del inmueble a consultar.
        :returns: Imagen de portada del inmueble consultado.
        """
        return self._fetch(f"{self.api_root}imagenes/portada/{inmueble_id}/{self.api_key}")

    def get_inmueble(self, inmueble_id: str) -> dict:
        """
        Obtiene el detalle de un inmueble por su ID.

        :param inmueble_id: ID del inmueble a consultar.
        :returns: Detalle del inmueble consultado.
        """
        return self._fetch(f"{self.api_root}inmuebles/get/{inmueble_id}/{self.api_key}")

    def list_inmuebles_by(self, empresa_id: str, tipo_inmueble: int, tipo_operacion: int, ciudad: int) -> dict:
        """
        Obtiene el listado de inmuebles filtrados por empresa, tipo de inmueble, tipo de operación y ciudad.

        :param empresa_id: ID de la empresa a consultar.
        :param tipo_inmueble: Código del tipo de inmueble (ej. casa, apartamento).
        :param tipo_operacion: Código del tipo de operación (ej. venta, renta).
        :param ciudad: Código de la ciudad a consultar.
        :returns: Lista de inmuebles filtrados.
        """
        return self._fetch(f"{self.api_root}inmuebles/listweb/{empresa_id}/{tipo_inmueble}/{tipo_operacion}/{ciudad}/{self.api_key}")

    def list_tipos_inmueble(self, empresa_id: str) -> dict:
        """
        Obtiene el listado de tipos de inmueble disponibles para una empresa.

        :param empresa_id: ID de la empresa a consultar.
        :returns: Lista de tipos de inmueble disponibles.
        """
        return self._fetch(f"{self.api_root}inmuebles/tiposInmueble/{empresa_id}/{self.api_key}")

    def list_tipos_operacion(self, empresa_id: str) -> dict:
        """
        Obtiene el listado de tipos de operación disponibles para una empresa.

        :param empresa_id: ID de la empresa a consultar.
        :returns: Lista de tipos de operación disponibles.
        """
        return self._fetch(f"{self.api_root}inmuebles/tiposOperacion/{empresa_id}/{self.api_key}")

    def list_ciudades(self, empresa_id: str) -> dict:
        """
        Obtiene el listado de ciudades disponibles para una empresa.

        :param empresa_id: ID de la empresa a consultar.
        :returns: Lista de ciudades disponibles.
        """
        return self._fetch(f"{self.api_root}inmuebles/ciudades/{empresa_id}/{self.api_key}")

    def list_inmuebles_portada(self, empresa_id: str) -> dict:
        """
        Obtiene el listado de inmuebles destacados para la portada de una empresa.

        :param empresa_id: ID de la empresa a consultar.
        :returns: Lista de inmuebles destacados para la portada.
        """
        return self._fetch(f"{self.api_root}inmuebles/portada/{empresa_id}/{self.api_key}")

    def list_inmuebles_recientes(self, empresa_id: str) -> dict:
        """
        Obtiene el listado de inmuebles recientes para una empresa.

        :param empresa_id: ID de la empresa a consultar.
        :returns: Lista de inmuebles recientes.
        """
        return self._fetch(f"{self.api_root}inmuebles/recientes/{empresa_id}/{self.api_key}")

    def get_inmueble_caracteristicas(self, inmueble_id: str) -> dict:
        """
        Obtiene el listado de características de un inmueble.

        :param inmueble_id: ID del inmueble a consultar.
        :returns: Detalle de las características del inmueble consultado.
        """
        return self._fetch(f"{self.api_root}caracteristicas/get/{inmueble_id}/{self.api_key}")

    def list_propiedades_inmueble(self, inmueble_id: str, tipo_propiedad: int) -> dict:
        """
        Obtiene el listado de propiedades de un inmueble.

        :param inmueble_id: ID del inmueble a consultar.
        :param tipo_propiedad: Código del tipo de propiedad a consultar (0, 1).
        :returns: Lista de propiedades del inmueble consultado.
        """
        return self._fetch(f"{self.api_root}properties/list/{inmueble_id}/{tipo_propiedad}/{self.api_key}")

    def get_proyecto(self, proyecto_id: str) -> dict:
        """
        Obtiene la información de un proyecto inmobiliario por su ID.

        :param proyecto_id: ID del proyecto a consultar.
        :returns: Detalle del proyecto consultado.
        """
        return self._fetch(f"{self.api_root}proyectos/get/{proyecto_id}/{self.api_key}")

    def list_proyectos(self, empresa_id: str) -> dict:
        """
        Obtiene el listado de proyectos de una empresa.

        :param empresa_id: ID de la empresa a consultar.
        :returns: Lista de proyectos de la empresa consultada.
        """
        return self._fetch(f"{self.api_root}proyectos/active/{empresa_id}/{self.api_key}")

    def get_proyecto_id_by_subdomain(self, subdomain: str) -> dict:
        """
        Obtiene el ID del proyecto basado en su subdominio.

        :param subdomain: Subdominio del proyecto a consultar.
        :returns: ID del proyecto asociado al subdominio.
        """
        return self._fetch(f"{self.api_root}proyectos/pidsubdir/{subdomain}/{self.api_key}")

    def get_proyecto_modelo(self, modelo_id: str) -> dict:
        """
        Obtiene el detalle de un modelo de inmueble dentro de un proyecto.

        :param modelo_id: ID del modelo a consultar.
        :returns: Detalle del modelo consultado.
        """
        return self._fetch(f"{self.api_root}modelos/get/{modelo_id}/{self.api_key}")

    def list_proyecto_modelos(self, proyecto_id: str) -> dict:
        """
        Obtiene el listado de modelos de un proyecto.

        :param proyecto_id: ID del proyecto a consultar.
        :returns: Lista de modelos del proyecto consultado.
        """
        return self._fetch(f"{self.api_root}modelos/active/{proyecto_id}/{self.api_key}")

    def get_setting(self, empresa_id: str, clave: str) -> dict:
        """
        Obtiene un valor de configuración específico de la empresa.

        :param empresa_id: ID de la empresa a consultar.
        :param clave: Clave de la configuración a consultar.
        :returns: Resultado de la operación y valor de la configuración consultada.
        """
        return self._fetch(f"{self.api_root}settings/get/{empresa_id}/{clave}/{self.api_key}")


if __name__ == "__main__":
    # Ejemplo de uso en un entorno local (fuera de un framework web)
    # Reemplaza con tus valores reales de API
    api_id = "TU_API_ID"
    api_key = "TU_API_KEY"

    inmostore_client = Website(api_id, api_key)

    try:
        inmostore_client.init()

        if inmostore_client.get_current_version():
            print(f"Versión de la API: {inmostore_client.get_current_version()}")

        # Ejemplo de una llamada a la API
        # Suponiendo que tienes un ID de empresa y de inmueble
        # inmueble_id_ejemplo = "ID_DEL_INMUEBLE_A_CONSULTAR"
        # inmueble = inmostore_client.get_inmueble(inmueble_id_ejemplo)
        # print(inmueble)

    except Exception as e:
        print(f"La aplicación falló: {e}")

File: java/classes/inmostore.client.java
// Archivo: InmostoreApiClient.java

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Esta clase implementa las funcionalidades del API de Portal Services
 * para web, adaptando las llamadas del plugin de Nuxt.js a un entorno Java.
 * Se encarga de la comunicación con la API desde el servidor.
 */
public class InmostoreApiClient {

    // Las siguientes constantes son para manejo de errores.
    public static final String ERROR_MESSAGE = "{\"Operacion\":{\"codigo\":-1, \"mensaje\":\"Error en la llamada a la API\"}, \"resultado\":null, \"items\":[]}";
    public static final String API_ROOT = "https://inmostore-api.psweb.me/";

    private String apiId;
    private String apiKey;

    /**
     * Constructor de la clase InmostoreApiClient.
     * @param apiId ID de la API (generalmente el ID de la empresa).
     * @param apiKey Clave de la API.
     */
    public InmostoreApiClient(String apiId, String apiKey) {
        System.out.println("InmoStore API Client for Java");
        this.apiId = apiId;
        this.apiKey = apiKey;

        // Si la clave de la API no está configurada, se muestra un error en la consola del servidor.
        if (this.apiKey == null || this.apiKey.isEmpty()) {
            System.err.println("*** No se ha detectado la configuración del sitio web. ***");
            // No se puede redirigir desde el servidor como en el JS, solo se imprime el error.
        }
    }

    /**
     * Helper para realizar una llamada GET a la API y devolver el resultado como String JSON.
     * @param url La URL completa de la API.
     * @return String JSON con el resultado de la llamada, o un JSON de error si falla.
     */
    private String getApiResponse(String url) {
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .build();
        try {
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            if (response.statusCode() == 200) {
                return response.body();
            } else {
                System.err.println("Error en la respuesta de la API: " + response.statusCode());
                return ERROR_MESSAGE;
            }
        } catch (IOException | InterruptedException e) {
            System.err.println("Error al conectar con el servidor de la API: " + e.getMessage());
            return ERROR_MESSAGE;
        }
    }

    /**
     * Obtiene el detalle de un elemento, consultado por su ID.
     * @param id ID del elemento a consultar.
     * @return String JSON con el detalle del elemento.
     */
    public String getElemento(int id) {
        String url = API_ROOT + "elementos/get/" + id + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el detalle de la empresa.
     * @param id ID de la empresa a consultar.
     * @return String JSON con el detalle de la empresa.
     */
    public String getEmpresa(String id) {
        String url = API_ROOT + "empresas/get/" + id + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene la lista de escenas activas de un objeto.
     * @param empresaId ID de la empresa a consultar.
     * @param modo Modo de la escena (0: Inmueble, 1: Proyecto).
     * @param objectId ID del objeto asociado a la escena.
     * @return String JSON con la lista de escenas.
     */
    public String listEscenas(String empresaId, int modo, String objectId) {
        String url = API_ROOT + "escenas/active/" + empresaId + "/" + modo + "/" + objectId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene la lista de pins de la escena actual.
     * @param escenaId ID de la escena a consultar.
     * @return String JSON con la lista de pins de la escena.
     */
    public String listEscenaPins(String escenaId) {
        String url = API_ROOT + "pins/list/" + escenaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene la URL de una imagen para embeber.
     * @param imageId ID de la imagen a consultar.
     * @return String con la URL de la imagen.
     */
    public String getEmbededImageUrl(String imageId) {
        return API_ROOT + "images/embed/" + imageId + "/" + this.apiKey;
    }

    /**
     * Obtiene el listado de imágenes de un objeto.
     * @param modo Modo de la imagen (0: Inmueble, 1: Proyecto).
     * @param objectId ID del objeto asociado a la imagen.
     * @return String JSON con la lista de imágenes.
     */
    public String listImagenes(int modo, String objectId) {
        String url = API_ROOT + "imagenes/list/" + modo + "/" + objectId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene la imagen de portada de un inmueble.
     * @param inmuebleId ID del inmueble a consultar.
     * @return String JSON con la imagen de portada.
     */
    public String getImagenPortada(String inmuebleId) {
        String url = API_ROOT + "imagenes/portada/" + inmuebleId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el detalle de un inmueble por su ID.
     * @param inmuebleId ID del inmueble a consultar.
     * @return String JSON con el detalle del inmueble.
     */
    public String getInmueble(String inmuebleId) {
        String url = API_ROOT + "inmuebles/get/" + inmuebleId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de inmuebles filtrados.
     * @param empresaId ID de la empresa.
     * @param tipoInmueble Tipo de inmueble.
     * @param tipoOperacion Tipo de operación.
     * @param ciudad Ciudad.
     * @return String JSON con la lista de inmuebles.
     */
    public String listInmueblesBy(String empresaId, int tipoInmueble, int tipoOperacion, int ciudad) {
        String url = API_ROOT + "inmuebles/listweb/" + empresaId + "/" + tipoInmueble + "/" + tipoOperacion + "/" + ciudad + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de tipos de inmueble disponibles para una empresa.
     * @param empresaId ID de la empresa.
     * @return String JSON con la lista de tipos de inmueble.
     */
    public String listTiposInmueble(String empresaId) {
        String url = API_ROOT + "inmuebles/tiposInmueble/" + empresaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de tipos de operación disponibles para una empresa.
     * @param empresaId ID de la empresa.
     * @return String JSON con la lista de tipos de operación.
     */
    public String listTiposOperacion(String empresaId) {
        String url = API_ROOT + "inmuebles/tiposOperacion/" + empresaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de ciudades disponibles para una empresa.
     * @param empresaId ID de la empresa.
     * @return String JSON con la lista de ciudades.
     */
    public String listCiudades(String empresaId) {
        String url = API_ROOT + "inmuebles/ciudades/" + empresaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de inmuebles destacados para la portada.
     * @param empresaId ID de la empresa.
     * @return String JSON con la lista de inmuebles.
     */
    public String listInmueblesPortada(String empresaId) {
        String url = API_ROOT + "inmuebles/portada/" + empresaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de inmuebles recientes.
     * @param empresaId ID de la empresa.
     * @return String JSON con la lista de inmuebles.
     */
    public String listInmueblesRecientes(String empresaId) {
        String url = API_ROOT + "inmuebles/recientes/" + empresaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de características de un inmueble.
     * @param inmuebleId ID del inmueble.
     * @return String JSON con las características del inmueble.
     */
    public String getInmuebleCaracteristicas(String inmuebleId) {
        String url = API_ROOT + "caracteristicas/get/" + inmuebleId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de propiedades de un inmueble.
     * @param inmuebleId ID del inmueble.
     * @param tipoPropiedad Tipo de propiedad.
     * @return String JSON con la lista de propiedades.
     */
    public String listPropiedadesInmueble(String inmuebleId, int tipoPropiedad) {
        String url = API_ROOT + "properties/list/" + inmuebleId + "/" + tipoPropiedad + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene la información de un proyecto inmobiliario por su ID.
     * @param proyectoId ID del proyecto.
     * @return String JSON con los detalles del proyecto.
     */
    public String getProyecto(String proyectoId) {
        String url = API_ROOT + "proyectos/get/" + proyectoId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de proyectos de una empresa.
     * @param empresaId ID de la empresa.
     * @return String JSON con la lista de proyectos.
     */
    public String listProyectos(String empresaId) {
        String url = API_ROOT + "proyectos/active/" + empresaId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el ID del proyecto basado en su subdominio.
     * @param subdomain Subdominio del proyecto.
     * @return String JSON con el ID del proyecto.
     */
    public String getProyectoIdBySubdomain(String subdomain) {
        String url = API_ROOT + "proyectos/pidsubdir/" + subdomain + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el detalle de un modelo de inmueble dentro de un proyecto.
     * @param modeloId ID del modelo.
     * @return String JSON con el detalle del modelo.
     */
    public String getProyectoModelo(String modeloId) {
        String url = API_ROOT + "modelos/get/" + modeloId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene el listado de modelos de un proyecto.
     * @param proyectoId ID del proyecto.
     * @return String JSON con la lista de modelos.
     */
    public String listProyectoModelos(String proyectoId) {
        String url = API_ROOT + "modelos/active/" + proyectoId + "/" + this.apiKey;
        return getApiResponse(url);
    }

    /**
     * Obtiene un valor de configuración específico de la empresa.
     * @param empresaId ID de la empresa.
     * @param clave Clave de la configuración.
     * @return String JSON con el valor de la configuración.
     */
    public String getSetting(String empresaId, String clave) {
        String url = API_ROOT + "settings/get/" + empresaId + "/" + clave + "/" + this.apiKey;
        return getApiResponse(url);
    }
}
